packages/react-components/src/components/MessageThread.tsx

```diff
@@ -715,6 +718,47 @@ export const MessageThreadWrapper = (props: MessageThreadProps): JSX.Element =>
     [onFetchAttachments]
   );
 
+  const localeStrings = useLocale().strings.messageThread;
+  const strings = useMemo(() => ({ ...localeStrings, ...props.strings }), [localeStrings, props.strings]);
+
+  const latestDeletedMessageIdRef = useRef<string | undefined>(undefined);
```

Comment:
Should this be able to handle multiple outstanding deletions?

Reply:
As we allow only 1 message id with `onDeleteMessage`, the messages array should always be up to date for a user and only single message can be deleted (otherwise, user will be able to delete the same message multiple times). So there should be only 1 deleted message as for now. We can always change this behavior in case if we will allow multiple messages to be deleted.